### ISSUE TITLE
Order of Iterators it not preserved

### ISSUE THREAD

**Issue by jecisc** [juliendelplanque/Iterators#20]
I am rewriting the code of a project to use iterators but it seems it breaks some of my tests.

Code:
```Smalltalk
clean
	self configuration localMethods iterator
		| [ :method | method selector = #initialize ] selectIt
		| [ :method | method ast nilAssignmentNodes isNotEmpty ] selectIt
		| [ :method | 'Do it 1' traceCr. method ast nilAssignmentNodes do: #removeFromTree ] doIt
		| [ :method | 'Do it 2' traceCr. method ast install ] doIt
		> Execute
```

I would expect in the transcript to get `Do it 1` before `Do it 2` but I get the opposite.

**Comment by juliendelplanque** [comment]
Let's investigate that.

**Comment by juliendelplanque** [comment]
I have a fix see #22
